File: RedSheet.DbEntities/Models/vGameRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vGameRecords", Schema = "dbo")]
	[Serializable]
	public partial class vGameRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int GameId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int GameTypeId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Play { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectGameDetailId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Trigger { get; set; } 

	public vGameRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-game-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vGameRecord {
    constructor(vGameRecord?: vGameRecord )  {
        let properties = [ "gameId", "gameTypeId", "play", "projectGameDetailId", "trigger",];
        for (let property of properties)
            if (vGameRecord && vGameRecord[property])
                this[property] = vGameRecord[property];
    }
 
	gameId : number =   0 ;
 
    @range(1,2147483647)
	gameTypeId : number =   undefined;
 
    @required()
    @maxLength(1000)
	play : string =   undefined;
 
    @range(1,2147483647)
	projectGameDetailId : number =   undefined;
 
    @required()
    @maxLength(1000)
	trigger : string =   undefined;


}
